File: HelpSoftware/Help/EF/ArticleProposal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Help
{
  public class ArticleProposal
  {
    [Key]
    public int ID_ArticleProposal { get; set; }

    public string ArticleTitle { get; set; }

    public string ArticleDescription { get; set; }

    public string Comment { get; set; }

    public ICollection<Topic> Topics { get; set; }

    public ICollection<Tag> Tags { get; set; }

    public DateTime CreatedDate { get; set; }

    public User AssignedUser { get; set; }

    public string RequesterName { get; set; }

    public string RequesterEMail { get; set; }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/ArticleProposal.ts
import { Topic, Tag, User } from "./EF";

export class ArticleProposal {
  ID_ArticleProposal: Number;
  ArticleTitle: string;
  ArticleDescription: string;
  Comment: string;
  Topics: Topic[];
  Tags: Tag[];
  CreatedDate: Date;
  AssignedUser: User;
  RequesterName: string;
  RequesterEMail: string;

  deserialize(input) {
    this.ID_ArticleProposal = input.ID_ArticleProposal;
    this.ArticleTitle = input.ArticleTitle;
    this.ArticleDescription = input.ArticleDescription;
    this.Comment = input.Comment;
    this.CreatedDate = input.CreatedDate;
    this.AssignedUser = input.AssignedUser;
    this.RequesterName = input.RequesterName;
    this.RequesterEMail = input.RequesterEMail;
    return this;
  }
}
